homeassistant/components/switchbot_cloud/vacuum.py

```diff
@@ -0,0 +1,135 @@
+"""Support for SwitchBot vacuum."""
+
+from typing import Any
+
+from switchbot_api import Device, Remote, SwitchBotAPI, VacuumCommands
+
+from homeassistant.components.vacuum import (
+    STATE_CLEANING,
+    STATE_DOCKED,
+    STATE_ERROR,
+    STATE_IDLE,
+    STATE_PAUSED,
+    STATE_RETURNING,
+    StateVacuumEntity,
+    VacuumEntityFeature,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import SwitchbotCloudData
+from .const import DOMAIN
+from .coordinator import SwitchBotCoordinator
+from .entity import SwitchBotCloudEntity
+
+
+async def async_setup_entry(
+    hass: HomeAssistant,
+    config: ConfigEntry,
+    async_add_entities: AddEntitiesCallback,
+) -> None:
+    """Set up SwitchBot Cloud entry."""
+    data: SwitchbotCloudData = hass.data[DOMAIN][config.entry_id]
+    async_add_entities(
+        _async_make_entity(data.api, device, coordinator)
+        for device, coordinator in data.devices.vacuums
+    )
+
+
+# vacuum state / workingStatus
+# "StandBy": "Stand by",
+# "Clearing": "Clearing",
+# "Paused": "Paused",
+# "GotoChargeBase": "Go to charge base",
+# "Charging": "Charging",
+# "ChargeDone": "Charge done",
+# "Dormant": "Dormant",
+# "InTrouble": "In trouble",
+# "InRemoteControl": "In remote control",
+# "InDustCollecting": "In dust collecting"
+
+# fan speed
+# "0": "Quiet",
+# "1": "Standard",
+# "2": "Strong",
+# "3": "MAX"
+VACUUM_SWITCHBOT_STATE_TO_HA_STATE: dict[str, str] = {
+    "StandBy": STATE_IDLE,
+    "Clearing": STATE_CLEANING,
+    "Paused": STATE_PAUSED,
+    "GotoChargeBase": STATE_RETURNING,
+    "Charging": STATE_DOCKED,
+    "ChargeDone": STATE_DOCKED,
+    "Dormant": STATE_IDLE,
+    "InTrouble": STATE_ERROR,
+    "InRemoteControl": STATE_CLEANING,
+    "InDustCollecting": STATE_DOCKED,
+}
+
+
+# https://github.com/OpenWonderLabs/SwitchBotAPI?tab=readme-ov-file#robot-vacuum-cleaner-s1-plus-1
+class SwitchBotCloudVacuum(SwitchBotCloudEntity, StateVacuumEntity):
+    """Representation of a SwitchBot vacuum."""
+
+    _attr_supported_features: VacuumEntityFeature = (
+        VacuumEntityFeature.BATTERY
+        | VacuumEntityFeature.FAN_SPEED
+        | VacuumEntityFeature.PAUSE
+        | VacuumEntityFeature.RETURN_HOME
+        | VacuumEntityFeature.START
+        | VacuumEntityFeature.STATE
+    )
+
+    _attr_name = None
+    _attr_fan_speed_list: list[str] = ["quiet", "standard", "strong", "max"]
```

Comment:
Let's put the list outside of the class

Reply:
We can add these fan speeds to the translations

Reply:
Hum, ok, but how to do it?  
Maybe in const.py? Or in other file?
And I suppose, same things for mapping state switchbot to HA?

Reply:
What's the best?  
- `_attr_fan_speed_list: list[str] = [VACUUM_FAN_SPEED_QUIET, VACUUM_FAN_SPEED_STANDARD, etc.]`
- or `_attr_fan_speed_list: list[str] = VACUUM_FAN_SPEED_LIST`

Reply:
And last question, I'm not sure how put translation.  
Could you give me a example or an other file with an usage, please?
Thanks in advance

Reply:
So, I use constant for each terme "quiet", "standard", "strong" and "max" and put it in const.py file  
I use a dict for mapping speed to speed waiting by switchbot  
  
For translation, I don't know how do it so, I wait for your response